web/consumers.py

```
@@ -0,0 +1,569 @@
+        except VirtualClassroom.DoesNotExist:
+            return False
+        except Exception as e:
+            logger.error(f"Error checking if user is teacher: {str(e)}")
+            return False
+
+    @database_sync_to_async
+    def save_canvas_data(self, canvas_data, background_image):
+        """Save canvas data to database"""
+        try:
+            classroom = VirtualClassroom.objects.get(id=self.classroom_id)
+            whiteboard, created = VirtualClassroomWhiteboard.objects.get_or_create(
```

Pylint (3.3.7)

[warning] Unused variable 'created'

(W0612)